src/roswire/ros2/launch.py

```diff
@@ -111,17 +112,23 @@ def locate(self, filename: str, *, package: Optional[str] = None) -> str:
         if not package:
             assert os.path.isabs(filename)
             return filename
-
-        filename_original = filename
-        app_description = self._app_instance.app.describe()
-        package_path = app_description.packages[package].path
-        filename = os.path.join(package_path, 'launch', filename_original)
-        if not self._app_instance.files.isfile(filename):
-            raise exc.LaunchFileNotFound(path=filename)
-        logger.debug('determined location of launch file'
-                     f' [{filename_original}] in package [{package}]: '
-                     f'{filename}')
-        return filename
+        else:
+            command = 'ros2 package prefix ' + shlex.quote(package)
```

Comment:
We should retrieve the package location from the package database to avoid a redundant and very expensive API call.

Reply:
I'm sorry, I'm not exactly sure how to do this. How do I access the package database?

Reply:
No worries. `self._app_instance` gives you access to the `AppInstance` to which the launch manager belongs. From there, you can get a description of the associated app:
```
app_description = self._app_instance.app.description
```

You can then get access to a description of the package from the `PackageDatabase`:
```
package_description = app_description.packages[package]
```

and finally:
```
package_location = package_description.path
```